internal/cluster/validations/validations.go

```diff
@@ -312,73 +315,298 @@ func ValidateVipDHCPAllocationWithIPv6(vipDhcpAllocation bool, machineNetworkCID
 	return nil
 }
 
-func ValidateIPAddresses(ipV6Supported bool, obj interface{}) error {
-	var allAddresses []*string
-	var apiVip string
-	var ingressVip string
-	var machineNetworks []*models.MachineNetwork
-	userManagedNetworking := false
-	vipDhcpAllocation := false
+func handleApiVipBackwardsCompatibility(clusterId strfmt.UUID, apiVip string, apiVips []*models.APIVip) ([]*models.APIVip, error) {
+	// APIVip provided, but APIVips were not.
+	if apiVip != "" && len(apiVips) == 0 {
+		return []*models.APIVip{{IP: models.IP(apiVip), ClusterID: clusterId}}, nil
+	}
```

Comment:
Similar to the comment at the beginning, I am not sure if you are required to carry the `ClusterID: clusterId` all the way down by yourself